packages/curve-ui-kit/src/features/connect-wallet/lib/ConnectionContext.tsx

```diff
@@ -58,7 +58,7 @@ const ConnectionContext = createContext<ConnectionContextValue<unknown>>({
 const compareSignerAddress = <TChainId extends any>(
   wallet: Wallet | null,
   lib: { chainId: TChainId; signerAddress?: string } | null,
-) => getWalletSignerAddress(wallet)?.toLowerCase() == lib?.signerAddress?.toLowerCase()
+) => !wallet || getWalletSignerAddress(wallet)?.toLowerCase() == lib?.signerAddress?.toLowerCase()
```

Comment:
`getWalletSignerAddress` already uses `?.` everywhere so not sure why this was necessary

Reply:
Hmm I'm not sure why this is necessary, the `llib?.signerAddress` should be null or undefined so using `==` should be enough. Anyway, for now I could explain what I meant

```suggestion
) =>
  // we assume the address is matched if there is no wallet connected
  !wallet || getWalletSignerAddress(wallet)?.toLowerCase() == lib?.signerAddress?.toLowerCase()
```